cmd/kubeadm/app/features/features.go

```diff
@@ -161,6 +161,16 @@ func NewFeatureGate(f *FeatureList, value string) (map[string]bool, error) {
 		featureGate[k] = boolValue
 	}
 
+	// Add any feature gates that haven't been specified, but default to true
+	for fgKey, fgValue := range *f {
+		_, exists := featureGate[fgKey]
+		if !exists {
+			if fgValue.Default {
+				featureGate[fgKey] = true
```

Comment:
@detiber 

Jason, wouldn't the above append all feature gates that the user didn't specify on the command line and set them to `true`? say if `DynamicKubeletConfig` was not set by the user this would override the default from `false` to `true`. 

please, correct me if i'm wrong.

Reply:
@neolit123 It will only set the ones that have a default value of `true`. Without this, updating the default value of the feature has no effect. That said, I'm not a fan of this hack, and will take another look today to see if there is a better way to bubble up the feature default value :)

Reply:
This is super weird... Why wouldn't we populate the struct from the flag itself?  

Is this an artifact of feature-gate mimicking.

Reply:
I've changed course on this and found a better fix that also does the proper thing for upgrade. I'm still working on fixing the plan and upgrade tests to handle it properly, though